code/obj/item/clothing/injector_mask_belt.dm

```diff
@@ -427,3 +406,153 @@ There's A LOT of duplicate code here, which isn't ideal to say the least. Should
+
+	for(var/A in cond_types)
+		var/datum/injector_belt_condition/C = new A()
+		. += C.name
+		if (setType)
+			.[C.name] = A
+
+/proc/autoinjector_ui_data(var/datum/reagents/R, var/datum/injector_belt_condition/condition, var/min_time, var/inj_amount)
+	. = list()
+
+	var/list/reagentData = null
+	if (R)
+		reagentData = list(
+			maxVolume = R.maximum_volume,
+			totalVolume = R.total_volume,
+			contents = list(),
+			finalColor = "#000000"
+		)
+
+		var/list/contents = reagentData["contents"]
+		if(istype(R) && R.reagent_list.len>0)
+			reagentData["finalColor"] = R.get_average_rgb()
+			for(var/reagent_id in R.reagent_list)
+				var/datum/reagent/current_reagent = R.reagent_list[reagent_id]
+
+				contents.Add(list(list(
+					name = reagents_cache[reagent_id],
+					id = reagent_id,
+					colorR = current_reagent.fluid_r,
+					colorG = current_reagent.fluid_g,
+					colorB = current_reagent.fluid_b,
+					volume = current_reagent.volume
+				)))
+	.["reagentData"] = reagentData
+
+	.["conditions"] = autoinjector_trigger_names(FALSE)
+
+	.["condition"] = null
+	.["conditionTreshold"] = null
+	.["conditionDamage"] = null
+	if (condition)
+		.["condition"] = list(
+			name = condition.name,
+			desc = condition.desc,
+		)
+
+		if (istype(condition, /datum/injector_belt_condition/with_threshold))
+			var/datum/injector_belt_condition/with_threshold/tresholdCondition = condition
+			.["conditionTreshold"]  = list(
+				currentValue = tresholdCondition.threshold,
+				suffix = tresholdCondition.suffix,
+				minValue = tresholdCondition.minValue,
+				maxValue = tresholdCondition.maxValue,
+			)
+
+			if (istype(condition, /datum/injector_belt_condition/with_threshold/damage))
+				var/datum/injector_belt_condition/with_threshold/damage/damageCondition = condition
+				.["conditionDamage"]  = list(
+					damagetype = damageCondition.damagetype
+				)
+
+	.["injectionAmount"] = inj_amount
+	.["minimumTime"] = min_time
+
+/proc/autoinjector_ui_act(obj/source, action, params, mob/user, var/maximum_volume)
+	var/obj/item/injector_belt/current_belt = null
+	var/obj/item/clothing/mask/gas/injector_mask/current_mask = null
+	if (istype(source, /obj/item/injector_belt))
+		current_belt = source
+	else if (istype(source, /obj/item/clothing/mask/gas/injector_mask))
+		current_mask = source
+
+	switch(action)
+		if ("remove_cont")
+			if (current_belt)
+				current_belt.container.set_loc(get_turf(source))
+				current_belt.container = null
+			else if (current_mask)
+				current_mask.container.set_loc(get_turf(source))
+				current_mask.container = null
+			. = TRUE
+
+		if ("remove_cond")
+			if (current_belt)
+				current_belt.condition = null
+			else if (current_mask)
+				current_mask.condition = null
+			. = TRUE
+
+		if ("sel_cond")
+			var/list/filtered = autoinjector_trigger_names(TRUE)
+
+			var/selected = params["condition"]
+			var/selected_type
+			var/datum/injector_belt_condition/selected_condition = null
+			if (selected <> "None")
```

Comment:
sorry, forgot to put this in the list of to-not-use in the style guide since i haven't seen it ever used in DM lol
```suggestion
			if (selected != "None")
```